.circleci/config.yml

```
@@ -177,7 +177,8 @@ jobs:
     steps:
       - setup_elixir-omg_workspace
       - run: make install-hex-rebar
-      - run: mix do compile --warnings-as-errors --force, credo, format --check-formatted --dry-run
+      - run: mix do compile --warnings-as-errors --force, credo --ignore-checks Credo.Check.Readability.SinglePipe, format --check-formatted --dry-run
+      - run: mix credo --only Credo.Check.Readability.SinglePipe | grep -c "Use a function call when a pipeline is only one function long" | awk '$1 > 369' 
```

How does this work? What this 369 mean?

---

Also why this needs to be separate from the command above?

---

Pretty clever!

@pgebal it's a few things.

* breaks apart the general credo routine to ignore single pipe checks
* runs the single pipe check separately, if its greater than the existing `369` occurrences of the single pipe, output the number

@InoMurko does this need to return explicitly return a failure exit code?

```suggestion
      - run: mix credo --only Credo.Check.Readability.SinglePipe | grep -c "Use a function call when a pipeline is only one function long" | awk '$1 > 369{exit 1}' 
```

---

Thanks Arthur. @InoMurko Why don't we want to fix single pipelines and stick to 1 run of credo? Is it because fixing all the single pipelines is too time consuming?

---

yeah, there's 369 occurrences at the moment. But we don't want to add new ones, and gradually fix those 369.